sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -474,7 +474,7 @@ object GpuCast extends Arm {
           }
         }
       case (StringType, dt: DecimalType) =>
-        castStringToDecimal(input, ansiMode, dt)
+      com.nvidia.spark.rapids.jni.CastStrings.toDecimal(input, ansiMode, dt.precision, dt.scale)
```

Comment:
`com.nvidia.spark.rapids.jni.CastStrings` can be imported then we just call `CastStrings.toDecimal`.